internal/api/appreciation.go

```diff
@@ -9,62 +9,68 @@ import (
 	"github.com/joshsoftware/peerly-backend/internal/app/appreciation"
 	"github.com/joshsoftware/peerly-backend/internal/pkg/apperrors"
 	"github.com/joshsoftware/peerly-backend/internal/pkg/dto"
+	log "github.com/joshsoftware/peerly-backend/internal/pkg/logger"
 	"github.com/joshsoftware/peerly-backend/internal/pkg/utils"
-	logger "github.com/sirupsen/logrus"
 )
 
 func createAppreciationHandler(appreciationSvc appreciation.Service) http.HandlerFunc {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 		var appreciation dto.Appreciation
 		err := json.NewDecoder(req.Body).Decode(&appreciation)
 		if err != nil {
-			logger.Errorf("Error while decoding request data : %v", err)
+			log.Errorf(req.Context(),"Error while decoding request data : %v", err)
```

Comment:
why request.Context() everywhere?

Reply:
created variable
ctx := req.Context()
and used it